textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/join/JoinTest.java

```diff
@@ -846,19 +848,537 @@ public void testIDFieldDoesNotExist() throws Exception {
+		bookTuple2 = new ArrayList<>(1);
+		bookTuple2.add(new DataTuple(bookSchema, book2));
+		
+		dataStoreForOuter = new DataStore(
+				DataConstants.INDEX_DIR + "/join_test_dir_1", bookSchema);
+		dataWriterForOuter = new DataWriter(dataStoreForOuter, analyzer);
+		dataStoreForInner = new DataStore(
+				DataConstants.INDEX_DIR + "/join_test_dir_2", bookSchema);
+		dataWriterForInner = new DataWriter(dataStoreForInner, analyzer);
+		dataWriterForOuter.clearData();
+		dataWriterForInner.clearData();
+		
+		dataWriterForOuter.writeData(bookTuple1);
+		dataWriterForInner.writeData(bookTuple2);
+		
+		KeywordPredicate keywordPredicate = null;
+		IDataStore dataStore = null;
+		IndexBasedSourceOperator indexInputOperator = null;
+		
+		String query = "special";
+		dataStore = dataStoreForOuter;
+		keywordPredicate = new KeywordPredicate(query, Arrays.asList(modBookAttr),
+				analyzer, DataConstants.KeywordMatchingType.CONJUNCTION_INDEXBASED);
+		indexInputOperator = new IndexBasedSourceOperator(keywordPredicate
+						.generateDataReaderPredicate(dataStore));
+		keywordMatcherOuter = new KeywordMatcher(keywordPredicate);
+		keywordMatcherOuter.setInputOperator(indexInputOperator);
+		
+		query = "writer";
+		dataStore = dataStoreForInner;
+		keywordPredicate = new KeywordPredicate(query, Arrays.asList(modBookAttr),
+				analyzer, DataConstants.KeywordMatchingType.CONJUNCTION_INDEXBASED);
+		indexInputOperator = new IndexBasedSourceOperator(keywordPredicate
+				.generateDataReaderPredicate(dataStore));
+		keywordMatcherInner = new KeywordMatcher(keywordPredicate);
+		keywordMatcherInner.setInputOperator(indexInputOperator);
+
+		List<ITuple> resultList = getJoinResults(keywordMatcherOuter,
+				keywordMatcherInner, idAttr, reviewAttr, 20);
+
+		Attribute[] schemaAttributes = { idAttr, authorAttr, pagesAttr, 
+				reviewAttr,	SchemaConstants.SPAN_LIST_ATTRIBUTE};
+
+		List<Span> spanList = new ArrayList<>();
+		String reviewField = reviewAttr.getFieldName();
+		
+		
+		Span span1 = new Span(reviewField, 11, 33, "special_writer",
+				"special kind of " + "writer");
+		spanList.add(span1);
+
+		IField[] book = { new IntegerField(52),  new StringField("Mary Roach"),
+				new IntegerField(288),
+				new TextField("It takes a special kind "
+						+ "of writer to make topics ranging from death to our "
+						+ "gastrointestinal tract interesting (sometimes "
+						+ "hilariously so), and pop science writer Mary Roach is "
+						+ "always up to the task."),
+				new ListField<>(spanList) };
+		ITuple expectedTuple = new DataTuple(new Schema(schemaAttributes),
+				book);
+		List<ITuple> expectedResult = new ArrayList<>(1);
+		expectedResult.add(expectedTuple);
+
+		boolean contains = TestUtils.containsAllResults(expectedResult,
+				resultList);
+
+		Assert.assertEquals(1, resultList.size());
+		Assert.assertTrue(contains);
+	}
+
+	// This case tests for the scenario when one of the attributes' (the one 
```

Comment:
"attributes'" -> "attributes".

It will be nice to write the schemas in a simple format in the comments.